aten/src/ATen/native/SpectralOps.cpp

```diff
@@ -993,4 +994,93 @@ Tensor istft(const Tensor& self, const int64_t n_fft, const optional<int64_t> ho
       onesidedOpt, lengthOpt, /*return_complex=*/false);
 }
 
+void _fft_fill_with_conjugate_symmetry_(const Tensor& input, IntArrayRef dim_) {
+  const auto input_sizes = input.sizes();
+  const auto input_strides = input.strides();
+  TORCH_CHECK(dim_.size() > 0);
+  DimVector dim(dim_.begin(), dim_.end());
+  at::maybe_wrap_dims(dim, input_strides.size());
+
+  if (input.numel() == 0 || input_sizes[dim.back()] <= 2) {
+    return;  // No elements need writing
+  }
+
+  // Small dimensions may be treated as batch dims since they don't get mirrored
+  dim.erase(
+      std::remove_if(dim.begin(), dim.end(), [&](int64_t dim) {
+        return (input_sizes[dim] <= 2);
+      }),
+      dim.end());
+
+  // Use TensorIterator to coalesce batch dimensions
+  // NOTE: Can't use TensorIterator loops because we need negative strides
+  auto iter = TensorIteratorConfig()
+      .add_output(input)
+      .add_input(input)
+      .resize_outputs(false)
+      .declare_static_shape(input_sizes, dim)
+      .build();
+
+  const auto iter_strides = iter.strides(0);
+  const auto iter_sizes = iter.shape();
+  const auto ndim = iter_strides.size() + dim.size();
+  DimVector in_strides(ndim), signal_half_sizes(ndim);
+
+  const auto element_size = iter.element_size(0);
+  auto* data_ptr = static_cast<char*>(input.data_ptr());
```

Comment:
Move this definition closer to the use of data_ptr on line 1044 below and make this pointer double const

Reply:
This can't be `const auto*` because `out_data` needs to be mutable.

Reply:
Ah, my mistake.